beacon-chain/p2p/fork_watcher.go

```diff
@@ -0,0 +1,30 @@
+package p2p
+
+import (
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/slotutil"
+)
+
+// A background routine which listens for new and upcoming forks and
+// updates the node's discovery service to reflect any new fork version
+// changes.
+func (s *Service) forkWatcher() {
+	slotTicker := slotutil.NewSlotTicker(s.genesisTime, params.BeaconConfig().SecondsPerSlot)
+	for {
+		select {
+		case currSlot := <-slotTicker.C():
+			currEpoch := helpers.SlotToEpoch(currSlot)
+			if currEpoch == params.BeaconConfig().AltairForkEpoch {
+				_, err := addForkEntry(s.dv5Listener.LocalNode(), s.genesisTime, s.genesisValidatorsRoot)
+				if err != nil {
+					log.WithError(err).Error("Could not add fork entry")
```

Comment:
A comment here would be helpful because it seems scary from reading this code that I could miss this exact conditional. What if my node is off right that epoch? What if I am online, turn off node before the fork epoch, and turn it on son after?

Reply:
If this is not a problem, then could be useful to tell the reader where else this will be initialized or covered

Reply:
> A comment here would be helpful because it seems scary from reading this code that I could miss this exact conditional. What if my node is off right that epoch? What if I am online, turn off node before the fork epoch, and turn it on son after?

That is fine, this logic is to specifically handle nodes which are running from an earlier epoch, so that their enr is updated with the correct metadata in the fork epoch.